crm-project/src/main/java/CRM/service/ItemService.java

```diff
@@ -85,11 +89,24 @@ public int delete(List<Long> ids) {
     }
 
     @Override
-    public Item update(long id, String field, String content) {
-        // checkIfExists
-        // make sure there is such a field in Item -> use reflection!
-
-        return null;
+    public Item update(UpdateObjectRequest updateObject, long itemId) throws NoSuchFieldException {
+        Item item = Validations.doesIdExists(itemId, itemRepository);
+        if(Validations.checkIfFieldIsCustomObject(updateObject.getFieldName())){
+            Object jpaRepo = Validations.getFieldObjectRepository(updateObject.getFieldName());
+            Object contentObj = Validations.doesIdExists(Long.valueOf((Integer) updateObject.getContent()), getRepoToUpdateField((Class) jpaRepo));
+            Validations.checkIfParentItemIsNotTheSameItem(updateObject.getFieldName(),itemId, Long.valueOf((Integer) updateObject.getContent()));
+            Validations.setContentToFieldIfFieldExists(item, updateObject.getFieldName(), contentObj);
+        }
+        else{
+            if(Validations.checkIfFieldIsNonPrimitive(updateObject.getFieldName())) {
+                LocalDateTime dueDate = LocalDateTime.now().plusDays(Long.valueOf((Integer) updateObject.getContent()));
+                Validations.setContentToFieldIfFieldExists(item, updateObject.getFieldName(), dueDate);
+            }
+            else{
+                Validations.setContentToFieldIfFieldExists(item, updateObject.getFieldName(), updateObject.getContent());
+            }
+        }
```

Comment:
same here, l think you should put this in a function, like the top "if" block.
Then you'll get:
if -> do this function,
else -> do this function
I promise it will look gorgeous!